### PR TITLE
[GLUTEN-1942][CH] Support function EqualNullSafe

### PR DESCRIPTION
**Pull request by exmy**
## What changes were proposed in this pull request?

(Fixes: \#1942)

## How was this patch tested?

UT

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala [hunk 1]
```diff
@@ -231,6 +231,14 @@ trait SparkPlanExecApi {
       attributeSeq: Seq[Attribute]): ExpressionTransformer =
     new NamedStructTransformerBase(substraitExprName, original, attributeSeq)
 
+  def genEqualNullSafeTransformer(
+      substraitExprName: String,
+      left: ExpressionTransformer,
+      right: ExpressionTransformer,
+      original: EqualNullSafe): ExpressionTransformer = {
+    new BinaryExpressionTransformer(substraitExprName, left, right, original)
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala:239]: Maybe this is not needed becausae Velox backend and CH can share the same transformer.

Comment (exmy) [reply gluten-core/src/main/scala/io/glutenproject/backendsapi/SparkPlanExecApi.scala]: Thanks for advise! Done.

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 2]
```diff
@@ -508,6 +508,13 @@ object ExpressionConverter extends SQLConfHelper with Logging {
           replaceWithExpressionTransformer(locate.start, attributeSeq),
           locate
         )
+      case equal: EqualNullSafe =>
+        BackendsApiManager.getSparkPlanExecApiInstance.genEqualNullSafeTransformer(
+          substraitExprName.get,
+          replaceWithExpressionTransformer(equal.left, attributeSeq),
+          replaceWithExpressionTransformer(equal.right, attributeSeq),
+          equal
```
[the remark this answers is not in the document]
Comment (exmy) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: I'm sorry, I didn't understand what 'process EqualNullSafe as EqualTo' means. Could you explain more?

[the remark this answers is not in the document]
Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: I meant just by using https://github.com/oap-project/gluten/pull/1951/files#diff-6c6142be53d757525c88d44ce7a8d96c80085a5527043f5dd878df647e988316R192. But looks like CH does not have a `equalnullsafe` function, is that the case?

[the remark this answers is not in the document]
Comment (exmy) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: Yes, CH does not have the `equalnullsafe ` function.

[the remark this answers is not in the document]
Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: Got that. Velox aready has that, so we still need different implementations here.

[the remark this answers is not in the document]
Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: Could you add a TODO here for Velox backend?

[the remark this answers is not in the document]
Comment (exmy) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: OK

**Review by rui-mo — APPROVED** [review]

**Comment by rui-mo** [comment]
@exmy Can we get this PR merged?

**Comment by exmy** [comment]
> @exmy Can we get this PR merged?

Yes, it's ok.

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**